### ISSUE TITLE
[Bug]: Upgrade @vitejs/plugin-react

### ISSUE THREAD

**Issue by xbc5** [storybookjs/storybook#25976]
### Describe the bug

The Nx generator (which storybook FORCES me to use) pulls `@storybook/react-vite@v7.6.13`, which has a dependency of `@vitejs/plugin-react@3.1.0` which has a peer dependency of `vite@4.1.0`, but Nx uses ~v5. I'd assume that a vite plugin REQUIRES the vite package.

In other words, storybook depends on v4, but Nx depends on v5.

plugin-react now supports v5 in the latest release.

Also, throwing when detecting an Nx workspace is a warning at best, so don't do that. It locks me into the Nx/storybook ecosystem without an option when something goes wrong.

**Comment by valentinpalkovic** [comment]
Duplicate of https://github.com/storybookjs/storybook/issues/25125

**Comment by h-bessa** [comment]
@xbc5 did you find a solution ?

**Comment by valentinpalkovic** [comment]
Storybook 8 doesn't have a dependency on @vitejs/plugin-reactanymore. Please update to the latest version of Storybook.
